Add option to set expiry date against clients

andrewhancox opened the pull request.

andrewhancox committed:
Add option to set expiry date against clients

djarran commented:
Hey Andrew, I've tested this on two local instances and this works just as expected. Thank you! We have just increased the version number to include the latest changes since the last version bump. 

Could you please microbump your version in `version.php` and `db/upgrade.xml` to reflect the latest version `2024030700` ?

andrewhancox commented:
> Hey Andrew, I've tested this on two local instances and this works just as expected. Thank you! We have just increased the version number to include the latest changes since the last version bump.
> 
> Could you please microbump your version in `version.php` and `db/upgrade.xml` to reflect the latest version `2024030700` ?

That's now done.

andrewhancox commented:
Has any thought been given to integrating this fix?

andrewhancox commented:
Any thoughts about integrating this?

bwalkerl commented:
Hi @andrewhancox ,

I haven't had a chance to look at this, but based on what @djarran has said we should be able to merge it if you resolve the conflicts again. Apologies for missing this earlier and making you do it twice.

andrewhancox commented:
Done - no problem on the wait, will be good to be fully aligned with the main version.

bwalkerl commented on version.php:
```diff
@@ -26,7 +26,7 @@
 $plugin->component = 'mod_scormremote';
-$plugin->release = 2024092700;
+$plugin->release = 2024092701;
 $plugin->version = 2024092700; // Keep in lockstep with version.
```
Just need to get CI green now. Looks like it's complaining about the version not matching release / upgrade steps.

andrewhancox replied:
Was rushing, should be sorted now.

bwalkerl commented on classes/form/client.php:
```diff
@@ -57,6 +57,8 @@ public function definition() {
         $mform->addRule('primarydomain', get_string('maximumchars', '', 255), 'maxlength', 255, 'server');
         $mform->addHelpButton('primarydomain', 'domain', 'mod_scormremote');
 
+        $mform->addElement('date_time_selector', 'expiry', get_string('expiry', 'mod_scormremote'), array('optional' => true));
```
One last thing - this string doesn't exist.

andrewhancox replied:
Missed that resolving the cherry pick... sorted now. Thanks for your patience!
